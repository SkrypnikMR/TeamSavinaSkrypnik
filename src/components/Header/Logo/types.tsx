import React from 'react';

import { ITheme } from 'src/components/App/types';

export interface IDiv extends ITheme{
    display: string;
    flexDirection: string;
    justifycontent: string;
    alignItems: string;
    width: string;
    fontFamily: string;
    color: string;
    fontSize: string;
    height: string;
    marginTop: string;
    cursor: string;
}
