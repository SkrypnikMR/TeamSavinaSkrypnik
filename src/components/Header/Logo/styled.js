import styled from 'styled-components';

export const StLogo = styled.div`
    display:flex;
    justify-content: space-evenly;
    align-items: center;
    width:30%;
    font-family: 'Play', sans-serif;
    color:white;
        h1{
            font-size: 30px;
        }
        img{
            height: 8vh;
        }
    cursor: pointer;
`;
