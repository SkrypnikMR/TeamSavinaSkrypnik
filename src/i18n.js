import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

import en from './locales/en.json';
import ar from './locales/ar.json';
import ru from './locales/ru.json';

const language = localStorage.getItem('lang') || 'en';

i18n.use(initReactI18next).init({
    resources: {
        en: { translations: en },
        ar: { translations: ar },
        ru: { translations: ru },
    },
    fallbackLng: ['en', 'ar', 'ru'],
    lng: language,
    debug: false,
    ns: ['translations'],
    defaultNS: 'translations',
    keySeparator: false,
    interpolation: {
        escapeValue: false,
        formatSeparator: ',',
    },
    react: {
        wait: true,
    },
});

export default i18n;
