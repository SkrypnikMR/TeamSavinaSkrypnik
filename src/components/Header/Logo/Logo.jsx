import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { StLogo } from './styled';

const Logo = ({ history, setValue }) => {
    const handleClick = () => {
        const { pathname } = history.location;
        if (pathname === '/' || pathname === '/registration') return history.push('/');
        setValue({ name: 'init', value: true });
        history.push('/chat');
    };
    const { t } = useTranslation();
    return (
        <StLogo onClick={handleClick}>
            <img src="./public/assets/images/logo.png" />
            <h1>{t('logo')}</h1>
        </StLogo>
    );
};

Logo.propTypes = {
    history: PropTypes.object,
    setValue: PropTypes.func,
};

export default Logo;
