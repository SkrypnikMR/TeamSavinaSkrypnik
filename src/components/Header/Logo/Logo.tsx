import React from 'react';
import { useTranslation } from 'react-i18next';
import Button from 'src/components/UI/Button';
import { StLogo } from './styled';
import { LOGO } from '../../UI/baseLayout';

const Logo = ({ history }) => {
    const goToGames = () => history.push('/mainPage');
    const goToStats = () => history.push('/statistics');
    const { t } = useTranslation();
    return (
        <StLogo >
            <img src={LOGO} onClick={goToGames} />
            <div>
                <Button
                    content={t('game_zone')}
                    fontSize='20px'
                    borderRadius="0px"
                    bgColor="transparent"
                    onClick={goToGames}
                />
                <Button
                    content={t('statistics')}
                    fontSize='20px'
                    borderRadius="0px"
                    bgColor="transparent"
                    onClick={goToStats}
                />
            </div>
        </StLogo>
    );
};

export default Logo;
