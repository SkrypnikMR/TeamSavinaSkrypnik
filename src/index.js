import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from '@redux-saga/core';
import rootReducer from './Store';
import App from './components/App';
import rootSaga from './Store/saga';
import './i18n';

const devTools = window?.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(saga),
    devTools,
  ),
);
saga.run(rootSaga);

render(
  <Provider store={store}>
    <App />
  </Provider >,
  document.getElementById('root'),
);
