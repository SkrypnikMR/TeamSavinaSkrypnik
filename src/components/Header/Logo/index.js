import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import Logo from './Logo.jsx';
import { setValue } from '../../../Store/user/actions';

const mapDispatchToProps = dispatch => ({ setValue: payload => dispatch(setValue(payload)) });

export default connect(null, mapDispatchToProps)(withRouter(Logo));
