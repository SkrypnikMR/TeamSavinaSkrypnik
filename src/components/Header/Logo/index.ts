import { withRouter } from 'react-router';

import Logo from './Logo';

export default withRouter(Logo);
