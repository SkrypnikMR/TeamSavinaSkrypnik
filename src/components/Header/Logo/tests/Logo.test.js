import React from 'react';
import { shallowSmart, mountSmart } from '/src/helpers/testHelper';
import Logo from '../Logo';
import '/src/i18n';
import 'react-i18next';

describe('Logo', () => {
    it('Should match snapshot', () => {
        const component = shallowSmart(<Logo />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render p', () => {
        const component = mountSmart(<Logo />);
        expect(component.find('img')).toHaveLength(1);
    });
    it('should render inputs', () => {
        const component = mountSmart(<Logo />);
        expect(component.find('h1')).toHaveLength(1);
    });
});
