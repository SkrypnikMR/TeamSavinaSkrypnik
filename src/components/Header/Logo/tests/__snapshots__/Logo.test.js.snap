// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Should match snapshot 1`] = `"<div class=\\"styled__StLogo-sc-9vy3sh-0 dOfUuH\\"><img src=\\"./public/assets/images/logo.png\\"/><h1>Header&amp;Shmeader</h1></div>"`;
